### ISSUE TITLE
Starting slinkity without react and react-dom installed results in an error

### ISSUE THREAD

**Issue by tylergaw** [Holben888/slinkity-home#6]
**Problem**

The [Quick start](https://slinkity.dev/docs/quick-start/) currently says that you should be able to start Slinkity with `npx slinkity --serve` after only installing the `slinkity` dependency, but doing so results in an error and does not build the site.

**Steps**
- `node v14.16.0`
- `npm v6.14.11`

- Starting with a new directory with nothing in it
- `npm init` follow prompts
- `npm i --save-dev slinkity`
- npx slinkity --serve

Main error message:
```
[11ty] Eleventy CLI Fatal Error: (more in DEBUG output)
```

Pertinent part of error message:

```
Cannot find module 'react-dom/server'
```

Full error output

```
[11ty] [11ty] Eleventy CLI Fatal Error: (more in DEBUG output)
[11ty] [11ty] > Error in your Eleventy config file '/Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/eleventy-wrapper.js'. You may need to run `npm install`.
[11ty] 
[11ty] `EleventyConfigError` was thrown
[11ty] [11ty] > Cannot find module 'react-dom/server'
[11ty] Require stack:
[11ty] - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugins/reactPlugin/toRendererHtml.js
[11ty] - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugins/reactPlugin/addShortcode.js
[11ty] - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugins/reactPlugin/index.js
[11ty] - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugin.js
[11ty] - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/eleventy-wrapper.js
[11ty] - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/TemplateConfig.js
[11ty] - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/TemplateRender.js
[11ty] - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/TemplateData.js
[11ty] - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/Eleventy.js
[11ty] - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/cmd.js
[11ty] 
[11ty] `Error` was thrown:
[11ty] [11ty]     Error: Cannot find module 'react-dom/server'
[11ty]     Require stack:
[11ty]     - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugins/reactPlugin/toRendererHtml.js
[11ty]     - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugins/reactPlugin/addShortcode.js
[11ty]     - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugins/reactPlugin/index.js
[11ty]     - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugin.js
[11ty]     - /Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/eleventy-wrapper.js
[11ty]     - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/TemplateConfig.js
[11ty]     - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/TemplateRender.js
[11ty]     - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/TemplateData.js
[11ty]     - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/src/Eleventy.js
[11ty]     - /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/cmd.js
[11ty]         at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
[11ty]         at Function.Module._load (internal/modules/cjs/loader.js:725:27)
[11ty]         at Module.require (internal/modules/cjs/loader.js:952:19)
[11ty]         at require (internal/modules/cjs/helpers.js:88:18)
[11ty]         at Object.<anonymous> (/Users/tyler/Desktop/slinkity-test/node_modules/slinkity/lib/plugins/reactPlugin/toRendererHtml.js:5:5)
[11ty]         at Module._compile (internal/modules/cjs/loader.js:1063:30)
[11ty]         at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
[11ty]         at Module.load (internal/modules/cjs/loader.js:928:32)
[11ty]         at Function.Module._load (internal/modules/cjs/loader.js:769:14)
[11ty]         at Module.require (internal/modules/cjs/loader.js:952:19)
[11ty] 0 exited with code 1
```

**Expectation**
Expected a site to build to `_site` and be available at `localhost:8080`

**Work around**

I see in the later part of the quickstart the instructions to install `react-dom`, but I'm guessing that's only meant to be required if you need React. But, to work around this error, you can install React and react dom with `npm i react react-dom --save-dev`

After that, everything builds as expected.

**Comment by ArkaneMoose** [comment]
Thanks so much for the detailed bug report! We still need to investigate further, but here's what I'm noticing preliminarily:

- In theory, `react` and `react-dom` should have automatically been installed in your project, as they are peer dependencies of `slinkity`.
- Interestingly, it looks like *none* of the peer deps got installed. For instance, 11ty should be installed locally, but it looks like it is running from npx's global cache instead.
  > /Users/tyler/.npm/_npx/46919/lib/node_modules/@11ty/eleventy/cmd.js
- My hunch is that this is related to the fact that you are using npm v6. We have been testing with npm v7. I'll try to reproduce the issue with your npm version.

**Comment by ArkaneMoose** [comment]
I was able to reproduce your issue! Indeed, npm v6 does not automatically install the `peerDependencies` of a package; [this was changed in npm v7:](https://github.blog/2020-10-13-presenting-v7-0-0-of-the-npm-cli/)

> - [Automatically installing peer dependencies](https://github.com/npm/rfcs/blob/latest/implemented/0025-install-peer-deps.md): prior to npm 7 developers needed to manage and install their own peer dependencies. The new peer dependency algorithm ensures that a validly matching peer dependency is found at or above the peer-dependent’s location in the node_modules tree.

The only thing npm v6 apparently does for peer dependencies is print warnings during installation:

```
npm WARN slinkity@0.0.3-alpha.5 requires a peer of react@^17.0.2 but none is installed. You must install peer dependencies yourself.
npm WARN slinkity@0.0.3-alpha.5 requires a peer of react-dom@^17.0.2 but none is installed. You must install peer dependencies yourself.
```

We should update our docs to emphasize the peer dependency requirement.

(Also, we should probably make `@11ty/eleventy` a peer dependency instead of a direct dependency, as this will allow the `npx` command to work as intended, and more reliably than before.)

**Comment by tylergaw** [comment]
Ohhh, I see. Thanks @ArkaneMoose. It's funny I've seen those npm peer deps warnings so many times I just ignore them now. Oops

> We should update our docs to emphasize the peer dependency requirement

I think that'd be great. Even a recommendation that people use npm 7 because of that would be cool. I'm on npm 6 still just because I haven't had a reason to update, but probably will now just because.

**Comment by Holben888** [comment]
@tylergaw Thanks again for the thorough issue! Instead of explaining different caveats by npm version, we've decided to simply remove the `react` + `react-dom` peer dependency altogether. It's totally valid to use Slinkity without React. So instead of forcing you to install it, we'll just wait for you use a `.jsx` file before we raise any exceptions 👍 

Also, **we're planning to archive this repository soon!** We moved the contents of this repo [to a `docs/` directory on the main Slinkity project](https://github.com/Holben888/slinkity/tree/main/docs). Expect to log future issues and feature requests over there.
